continuous-async-profiler/src/main/java/com/github/krzysztofslusarski/asyncprofiler/ContinuousAsyncProfiler.java

```diff
@@ -62,6 +73,6 @@ private void createOutputDirectories(ContinuousAsyncProfilerProperties propertie
     @Override
     public void destroy() {
         log.info("Spring context destroyed, shutting down threads");
-        threads.forEach(Thread::interrupt);
+        scheduledFutures.forEach(scheduledFuture -> scheduledFuture.cancel(true));
```

Comment:
I guess we should check here if the last profiling is still working if so we should stop it.